src/main/kotlin/org/elm/workspace/commandLineTools/ElmReviewStreamModel.kt

```diff
@@ -0,0 +1,285 @@
+    // TODO Add a optional fix field (For later)
+    var suppressed: Boolean? = null,
+    var path: String? = null,
+    var rule: String? = null,
+    var message: String? = null,
+    var regionWatch: RegionWatch? = null,
+    var html: String? = null
+)
+
+data class RegionWatch(
+    var start: LocationWatch? = null,
+    var end: LocationWatch? = null
+)
+
+data class LocationWatch(
+    var line: Int = 0,
+    var column: Int = 0
+)
+
+sealed class ChunkWatch {
+    data class Unstyled(var str: String) : ChunkWatch()
+    data class Styled(var string: String? = null, var bold: Boolean? = null, var underline: Boolean? = null, var color: String? = null, var href: String? = null) : ChunkWatch()
+}
+
+enum class ReviewOutputType(val label: String) {
+    ERROR("error"), COMPILE_ERRORS("compile-errors"), REVIEW_ERRORS("review-errors");
+
+    companion object {
+        fun fromLabel(label: String) =
+            when (label) {
+                ERROR.label -> ERROR
+                COMPILE_ERRORS.label -> COMPILE_ERRORS
+                REVIEW_ERRORS.label -> REVIEW_ERRORS
+                else -> TODO("unknown type $label")
+            }
+    }
+}
+
+fun parseReviewJsonStream(reader: JsonReader, process: Process, emit: (List<ElmReviewWatchError>) -> Unit): Int {
+    with(reader) {
+        while (process.isAlive) {
+            if (process.errorStream.available() > 0)
+                return 1
+            if (hasNext()) {
+                val errors = readErrorReport()
+                emit(errors)
+            }
+        }
+        close()
+        return process.exitValue()
+    }
+}
+
+fun JsonReader.readErrorReport(): List<ElmReviewWatchError> {
+    val errors = mutableListOf<ElmReviewWatchError>()
+    if (this.peek() == JsonToken.END_DOCUMENT) return emptyList()
+    beginObject()
+    var type: ReviewOutputType? = null
+    while (hasNext()) {
+        when (nextName()) {
+            "type" -> {
+                type = ReviewOutputType.fromLabel(nextString())
+            }
+            "errors" -> {
+                when (type) {
+                    ReviewOutputType.REVIEW_ERRORS -> {
+                        beginArray()
+                        while (hasNext()) {
+                            var currentPath: String? = null
+                            beginObject()
+                            while (hasNext()) {
+                                when (nextName()) {
+                                    "path" -> currentPath = nextString()
+                                    "errors" -> {
+                                        beginArray()
+                                        while (hasNext()) {
+                                            val elmReviewWatchError = ElmReviewWatchError(path = currentPath)
+                                            beginObject()
+                                            while (hasNext()) {
+                                                when (nextName()) {
```

Comment:
Can't say I'm a huge fan of this, but whatever works 🤷

Reply:
yeah, me neither :)
but if we can find a decent stream parsing helper, I'd be glad to change that

Reply:
for unnesting I could try to factor out some of the nesting with the current parser

Reply:
not a blocker for merging this PR.